### PR TITLE
fix:refactor: getNumThreads return the omp_num_threads enabled and add some testcases

### PR DESCRIPTION
**Pull request by i8run**

### CODE REVIEW

**Comment by qiuxin2012** [comment]
LGTM

File: mkl2017-xeon-blas/mkl/src/main/c/jni/mkl.c [hunk 1]
```diff
@@ -33,17 +34,17 @@ JNIEXPORT void JNICALL Java_com_intel_analytics_bigdl_mkl_MKL_disableFastMM
  */
 JNIEXPORT void JNICALL Java_com_intel_analytics_bigdl_mkl_MKL_setBlockTime
   (JNIEnv * env, jclass cls, jint time) {
-  kmp_set_blocktime(0);
+  kmp_set_blocktime(time);
 }
 
 /*
  * Class:     com_intel_analytics_bigdl_mkl_watiPolicyPassive
```
Comment (yiheng) [inline mkl2017-xeon-blas/mkl/src/main/c/jni/mkl.c]: change the name in the doc?

File: mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java [hunk 2]
```diff
@@ -53,35 +54,65 @@
...
-        setNumThreads(mklNumThreads);
-        setBlockTime(mklBlockTime);
-        waitPolicyPassive();
+    public static int getMklWaitPolicy() {
+        String mklWaitPolicy = System.getProperty("bigdl.mklWaitPolicy", "passive").toLowerCase();
 
-        if (mklDisableFastMM) {
-            disableFastMM();
+        if (mklWaitPolicy.equalsIgnoreCase("passive")) {
+            return 3;
```
Comment (yiheng) [inline mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java]: define some static const variable to give the number with a name 

final static int MKL_WAIT_POLICY_PASSIVE = 3
final static int MKL_WAIT_POLICY_ACTIVE = 2

do not use magic number

File: mkl2017-xeon-blas/mkl/src/main/c/jni/mkl.c [hunk 3]
```diff
@@ -33,17 +34,17 @@ JNIEXPORT void JNICALL Java_com_intel_analytics_bigdl_mkl_MKL_disableFastMM
...
  * Method:    setNumThreads
```
Comment (yiheng) [inline mkl2017-xeon-blas/mkl/src/main/c/jni/mkl.c]: change setNumThreads?

File: mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java [hunk 4]
```diff
@@ -17,6 +17,9 @@
     private static boolean isLoaded = false;
     private static File tmpFile = null;
 
+    private final static int MKL_WAIT_POLICY_PASSIVE = 3;
```
Comment (yiheng) [inline mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java]: it should be public, right?

File: mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java [hunk 5]
```diff
@@ -17,6 +17,9 @@
...
+    private final static int MKL_WAIT_POLICY_ACTIVE = 2;
```
Comment (yiheng) [inline mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java]: it should be public, right?

**Comment by yiheng** [comment]
LGTM

**Pull request merged**